.github/workflows/build-firmware.yaml

```diff
@@ -298,8 +298,17 @@ jobs:
       run: echo '::error::Branch names must not contain ".", this breaks firmware autoupdates.' && exit 1
     - name: Execution throttle early exit
       # Don't skip any jobs if this workflow was run manually.
-      if: ${{ matrix.skip-rate && github.event_name != 'workflow_dispatch' }}
-      run: if (($(($RANDOM % 100)) < ${{ matrix.skip-rate }})); then echo "skip=true" >> $GITHUB_ENV; fi
+      if: ${{ matrix.skip-rate && github.event_name != 'workflow_dispatch' || contains(github.event.head_commit.message, 'only:') }}
```

Comment:
our yaml is getting more and more complicated. please add usage documentation and/or example as a code comment right into yaml?